specify/specify7
# Handle case when default formtable requires default view

Fixes #4749 

> Thankfully there is some consistency amongst the table-only views. Notably, every `altview` in the `altviews` of the view points to a view definition which is of type `formtable`. When reading a form definition for a subview on the frontend, if every altview points to a `formtable` viewdef in a view, we can likely fallback to using the `table.view` view for rendering the expanded definition of the subview (the collapsed view can still remain the original view).

Alternative implementation of #4760 which follows the above proposed solution from https://github.com/specify/specify7/pull/4760#issuecomment-2046022176

### Checklist

      and self-explanatory (or properly documented)

### Testing instructions
- In the instance, make sure there are no custom views defined for Collectors
- On the view definition for the CollectingEvent, make sure the `collectors` relationship is using the `Collectors` view (ignore this step if using the default): 
```xml
<cell type="subview" viewname="Collectors" id="collect" name="collectors" colspan="13" defaulttype="table" rows="3"/>
```
- Ensure the Collector subview correctly renders the default `Collector` view (pictured below) when the Subview is: 
  - [ ] In Subform Mode
  - [ ] In Grid Mode and the cell is expanded
  - [ ] Also ensure the correct collapsed view is shown when the Subview is in Grid Mode and the cell is collapsed
- On the CollectingEvent form definition, modify the viewname for the `collectors` relationship to be `Collector`
  - [ ] Ensure the collapsed view when the Subview is in Grid Mode is using the `Collector` view definition

- [ ] On the default Agent form, ensure the GroupPerson/Members subview appears as it did in `v7.9+`

---
**Default Collector View form:** 

<img width="784" alt="Screenshot 2024-04-11 at 10 56 06 AM" src="https://github.com/specify/specify7/assets/64045831/d25d4b54-4e65-4cad-86cd-83672b656b31">


**Default collapsed Collector View:** 

<img width="679" alt="Screenshot 2024-04-11 at 10 58 28 AM" src="https://github.com/specify/specify7/assets/64045831/74699bca-6596-461a-9627-436892934012">

**Collapsed view definition for `Collector` View:**

<img width="678" alt="Screenshot 2024-04-11 at 11 18 47 AM" src="https://github.com/specify/specify7/assets/64045831/83078b76-2aa7-4336-81c2-caeab7867c23">
